plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -614,7 +634,7 @@ def _decide_rpms_to_download(self, metadata_files, catalog):
             # scan through all the metadata to decide which packages to download
             package_info_generator = packages.package_list_generator(
                 primary_file_handle, primary.PACKAGE_TAG, primary.process_package_element)
-            wanted = self._identify_wanted_versions(package_info_generator)
+            wanted = self._identify_wanted_versions(package_info_generator, ids.TYPE_ID_RPM)
```

Comment:
I'll describe another option, besides the one I suggest below about just raising the exception inside the ``_identify_wanted_versions`` method. I'm not sure which I like better, and maybe you'll have other ideas entirely.

It feels weird to be passing in the type, when each item in the generator is already typed, and to have this side effect where the method stores a count in an instance attribute.

You could instead make no changes to the ``_identify_wanted_versions()`` method, and count the items in the generator with a little wrapper. It would go something like this:

```
counter = itertools.count()
wrapped_generator = itertools.imap(lambda x, y: x, package_info_generator, counter)
wanted = self._identify_wanted_versions(wrapped_generator)
total_rpm_count = counter.next()
```

Then as with other suggestions, you could do the comparison right here and raise an exception if necessary.

What do you think?